### ISSUE TITLE
Define some sort of api to indicate your peer.

### ISSUE THREAD

**Issue by adriancole** [opentracing/opentracing.github.io#18]
A common example in #14 is a discussion about integer to string conversion in relationship with capturing peer endpoint information portably. This is only an issue, if we expect to attribute a peer via a loose collection of KV pairs. (aka duck typing).

If endpoint information was recorded in a single value (associated with a peer, for example), we'd unload this quite a bit, and permit some existing practice to occur.. ex.
* HTrace passes context by reference, not by value `process_id`
* GRPC passes endpoint context by an opaque string `census_set_rpc_client_peer`
* zipkin passes this via a struct (which will eventually need to be extended to ipv6) `Endpoint`
* can dig further if folks are unconvinced..

Moreover, users don't often explicitly pass peer or local context.. frameworks do. In other words, this is a background concern for users more often than not.

Libraries and frameworks can *easily* deal with inefficiencies, for example caching any conversions if somehow this has been benchmarked to be a critical path concern.

Finally, the fields marked "peer", which often enough indicate a host or similar, could be common enough to warrant a conversion helper. Ex. I could easily write a codec from a base type to a host. Ex in some systems, that would be looking up the "process id" datastore. In others, maybe parsing an Alt-Svc header format, or in google cloud trace, parsing a URI.

Basically, I think we can unload a lot of chatter about itoa etc and also answer a fairly common set of use cases for distributed tracing, by explicitly calling out local and peer in a way that doesn't require it to be collected from several key/values mixed in with "foo"="bar".

*Note* none of this precludes an implementation from identifying the host value as a map of k/v pairs. ex. `setPeer(kv_pairs)` vs `setPeer(encoded_string)` vs `setPeer(struct)` could be choices.

**Comment by adriancole** [comment]
PS since it isn't easy to express conviction level.. My opinion on this isn't strongly held. It is primarily driven by a desire to reduce performance-related distractions #14.

**Comment by yurishkuro** [comment]
Adrian, I am not quite clear what you're suggesting - to extend the API with methods like `setPeer`?  We didn't want to go that way because it would be too easy to get stuck on trying to agree what the exact format of the arguments to such method would be. The arguments need to be unambiguously defined since the same instrumentation code would need to work with all possible tracers.

**Comment by adriancole** [comment]
My suggestion is to move the topic of describing peer, which as you mentioned isn't universally supported, from the setTag api to someplace else, such as setPeer or similar. This contains that discussion to a place where those who actually use it can participate, and those who don't can safely ignore the topic.

**Comment by bensigelman** [comment]
http://opentracing.io/data-semantics/#peer-tags

**Comment by bensigelman** [comment]
(feel free to reopen or speak up on Gitter if there's an objection to closing this)
